### PR TITLE
Apply design feedback on Metrics and DistributedContext

### PR DESCRIPTION
**Pull request by tarekgh**
- Reduce the number of abstract methods in Meter and allow overriding protected methods to avoid make it public.
- Rename the word “Long” in the APIs to “Int64”. This is the design guidelines recommendation.
- Add convenient properties NoPropagationEntry and UnlimitedPropagationEntry in EntryMetadata.
- Avoid using the 3 uppercase letters acronym of TTL and just use the full words TimeToLive.

### CODE REVIEW

**Comment by tarekgh** [comment]
@lmolkova @SergeyKanzhelev @cijothomas could you please have a quick look? Thanks.

File: src/OpenTelemetry.Api/DistributedContext/EntryMetadata.cs [hunk 1]
```diff
@@ -22,27 +22,37 @@ namespace OpenTelemetry.Context
...
         /// <summary>
         /// Initializes a new instance of the <see cref="EntryMetadata"/> struct.
         /// </summary>
-        /// <param name="entryTTL">TTL for the distributed context entry.</param>
-        public EntryMetadata(int entryTTL)
+        /// <param name="timeToLive">TTL for the distributed context entry.</param>
+        public EntryMetadata(int timeToLive)
         {
-            this.EntryTTL = EntryMetadata.NoPropagation;
+            this.TimeToLive = EntryMetadata.NoPropagation;
```
[the remark this answers is not in the document]
Comment (tarekgh) [reply src/OpenTelemetry.Api/DistributedContext/EntryMetadata.cs]: good catch and good point. I'll fix that. I'll mark the construct as private too for now and in the future we can just expose it if we run into more needed cases.

File: src/OpenTelemetry.Api/DistributedContext/EntryMetadata.cs [hunk 2]
```diff
@@ -22,27 +22,37 @@ namespace OpenTelemetry.Context
...
-        public int EntryTTL { get; }
+        public static EntryMetadata NoPropagationEntry => new EntryMetadata(NoPropagation);
+
+        /// <summary>
+        /// Gets a new instance of the <see cref="EntryMetadata"/> struct with UnlimitedPropagation value.
+        /// </summary>
+        public static EntryMetadata UnlimitedPropagationEntry => new EntryMetadata(UnlimitedPropagation);
+
+        /// <summary>
+        /// Gets the TimeToLive is either NO_PROPAGATION (0) or UNLIMITED_PROPAGATION (-1).
```
[the remark this answers is not in the document]
Comment (tarekgh) [reply src/OpenTelemetry.Api/DistributedContext/EntryMetadata.cs]: will do that. thanks.

**Comment by tarekgh** [comment]
I think the failures here are not related to my changes. Could someone take a look?


```
[xUnit.net 00:00:03.72]     OpenTelemetry.Trace.Export.Test.BatchingSpanProcessorTest.ProcessorDoesNotBlockOnExporter [FAIL]
  X OpenTelemetry.Trace.Export.Test.BatchingSpanProcessorTest.ProcessorDoesNotBlockOnExporter [1s 44ms]
  Error Message:
   Expected at least 1, got 0
Expected: True
Actual:   False
  Stack Trace:
     at OpenTelemetry.Trace.Export.Test.BatchingSpanProcessorTest.WaitForSpans(TestExporter exporter, Int32 spanCount, TimeSpan timeout) in d:\a\1\s\test\OpenTelemetry.Tests\Impl\Trace\Export\BatchingSpanProcessorTests.cs:line 341
   at OpenTelemetry.Trace.Export.Test.BatchingSpanProcessorTest.ProcessorDoesNotBlockOnExporter() in d:\a\1\s\test\OpenTelemetry.Tests\Impl\Trace\Export\BatchingSpanProcessorTests.cs:line 243
```

**Comment by lmolkova** [comment]
> I think the failures here are not related to my changes. Could someone take a look?

Agree. Test if flaky. Could you create an issue - I'll fix it later?

**Comment by tarekgh** [comment]
created https://github.com/open-telemetry/opentelemetry-dotnet/issues/344

**Pull request merged**